sql/src/main/java/io/crate/analyze/relations/TableRelation.java

```diff
@@ -246,4 +264,58 @@ public Symbol visitReference(Reference symbol, TableRelation context) {
             return null;
         }
     }
+
+    private static class GroupByValidator extends TraversalSymbolVisitor {
+
+        @Override
+        public Void visitReference(Reference symbol, TableRelation context) {
+            if (symbol.info().indexType() == ReferenceInfo.IndexType.ANALYZED) {
+                throw new IllegalArgumentException(
+                        String.format("Cannot GROUP BY '%s': grouping on analyzed/fulltext columns is not possible",
+                                SymbolFormatter.format(symbol)));
+            } else if (symbol.info().indexType() == ReferenceInfo.IndexType.NO) {
+                throw new IllegalArgumentException(
+                        String.format("Cannot GROUP BY '%s': grouping on non-indexed columns is not possible",
+                                SymbolFormatter.format(symbol)));
+            }
+            return null;
+        }
+    }
+
+    private static class OutputValidatorContext {
+        private boolean insideAggregation = false;
+    }
+
+    private static class OutputValidator extends SymbolVisitor<OutputValidatorContext, Void> {
+
+        @Override
+        public Void visitFunction(Function symbol, OutputValidatorContext context) {
+            context.insideAggregation = context.insideAggregation || symbol.info().type().equals(FunctionInfo.Type.AGGREGATE);
+            for (Symbol argument : symbol.arguments()) {
+                process(argument, context);
+            }
+            context.insideAggregation = false;
+            return null;
+        }
+
+        @Override
+        public Void visitReference(Reference symbol, OutputValidatorContext context) {
+            if (context.insideAggregation) {
+                ReferenceInfo.IndexType indexType = symbol.info().indexType();
+                if (indexType == ReferenceInfo.IndexType.ANALYZED) {
+                    throw new IllegalArgumentException(String.format(
+                            "Cannot select analyzed column '%s' within grouping or aggregations", SymbolFormatter.format(symbol)));
```

Comment:
this should be handled in the consumer. since this is just lack of implementation